ctru-rs/src/services/ac.rs

```diff
@@ -0,0 +1,331 @@
+use std::ffi::CString;
+use std::io::Write;
+use std::marker::PhantomData;
+pub struct Ac(());
+
+impl Ac {
+    /// Initialize a new service handle.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// #
+    /// use ctru::services::ac::Ac;
+    ///
+    /// let ac = Ac::new()?;
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "acInit")]
+    pub fn new() -> crate::Result<Ac> {
+        unsafe {
+            ResultCode(ctru_sys::acInit())?;
+            Ok(Ac(()))
+        }
+    }
+
+    /// Waits for an internet connection
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// #
+    /// use ctru::services::ac::Ac;
+    ///
+    /// let ac = Ac::new()?;
+    ///
+    /// println!("Waiting for an internet connection...");
+    /// ac.wait_for_internet_connection()?;
+    /// println!("Connected.");
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "acWaitInternetConnection")]
+    pub fn wait_internet_connection(&self) -> crate::Result<()> {
+        unsafe {
+            ResultCode(ctru_sys::acWaitInternetConnection())?;
+
+            Ok(())
+        }
+    }
+
+    /// Returns whether the console is connected to Wi-Fi
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// #
+    /// use ctru::services::ac::Ac;
+    ///
+    /// let ac = Ac::new()?;
+    ///
+    /// println!("Wi-Fi connected: {}", ac.get_wifi_status()?);
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "ACU_GetWifiStatus")]
+    pub fn get_wifi_status(&self) -> crate::Result<bool> {
+        unsafe {
+            let mut ret = 0u32;
+            ResultCode(ctru_sys::ACU_GetStatus(&mut ret))?;
+
+            Ok(ret == 3)
+        }
+    }
+
+    /// Returns whether the console is connected to Wi-Fi
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// #
+    /// use ctru::services::ac::Ac;
+    ///
+    /// let ac = Ac::new()?;
+    ///
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "ACU_GetWifiSecurityMode")]
+    pub fn get_wifi_security(&self) -> crate::Result<SecurityMode> {
+        unsafe {
+            let mut ret = 0u32;
+            ResultCode(ctru_sys::ACU_GetSecurityMode(&mut ret))?;
+            // fix this, for some reason the bindings have the type as a struct and not enum
+            // and so i can't impl TryFrom automatically
```

Comment:
Hmm, strange, on my system it looks like it's just a `libc::c_uint` alias? What does the struct definition look like? I wonder if we might just need to adjust some flags in `ctru-sys/build.rs` to change how the bindings are generated...

Reply:
here is the enum definition

https://github.com/devkitPro/libctru/blob/bf4a24a4e7e3e9fa1642845818d3d59ccb8c506a/libctru/include/3ds/services/ac.h#L17C6-L8C6

i really don't know why it's just a `c_uint` alias.